src/amo/reducers/users.js

```diff
@@ -60,6 +62,7 @@ export type UsersStateType = {
   currentUserID: UserId | null,
   byID: { [userId: UserId]: UserType },
   byUsername: { [username: string]: UserId },
+  isEditingById: { [userId: UserId]: boolean },
```

Comment:
This is similar to what we just discussed about a flag for a collection being edited. Why do we need to store multiple objects that represent multiple users being edited, as opposed to just a single flag that says "a user is being edited"?

Reply:
For right now I guess we don't, although editing multiple users is a thing I could see happening.

I guess actually, to your point... editing multiple collections could happen too in the future.

Maybe what we want to do is have a special key for a new/unexisting collection/user. Sorry, that sucks because I said you could change it but now I'm stepping back.

That said we ALWAYS have a user id because by the time you're signed in/editing on the front-end the server made you a user. So this way at least in the future we could allow multiple user edits at once.

I guess if I did it the other way I wouldn't say we should change it, but now that this one works and there's no drawback I don't know if it's worth changing? What does it buy us other than a *very* marginally simpler code path? Keying by ID might back debugging easier in some bizarre scenario.

Sorry, I'm clearing stepping back from my earlier suggestion of not caring re: collection ID keys.

Reply:
How would editing multiple users or collections work? Are you thinking a grid of forms where multiple forms for multiple users/collections would be displayed on a single screen?

Reply:
Yeah, for an admin thing it could happen. That said it’s nowhere in the mocks.

I guess the questions are:

1. Is the current code here significantly harder-to-follow than a single `isEditing` flag?
2. Is refactoring to add userIds later very difficult/annoying?
3. Are we likely to ever want multiple user edits?

I think the answer to 1. is no, 2. is yes, 3. is no.

So I'll change it if you feel strongly about it. I guess it'd be a smaller patch/reducer.

Reply:
I agree with your answers to the questions, so either way seems fine. I guess for me I tend to never add code that isn't going to be used - you know the old YAGNI chestnut. I might even suggest that the answer to 2. is yes, but not actually *very* difficult - just a pain.

In this user case it doesn't really matter, but with the collections one, not having it keyed avoids the issue of what to use for the key if there is no existing collection id/slug.

Your call. 😄